### PR TITLE
Optimize performance in test262

### PR DESCRIPTION
**Pull request by rPraml**
This PR targets two issues I found with a profiler, while running the 262 test suite:

- The method `addTestFiles` has a nested loop with a complexity of O(n^2). I've replaced this with a hashmap, that improves startup time from ~3 sec down to <1 sec
- The getSourcePositionFromStack uses the more efficient StackWalker now

### CODE REVIEW

**Review by rbri — APPROVED** [review]
LGTM

**Comment by rbri** [comment]
NativeJavaListTest > length FAILED
    org.junit.ComparisonFailure at NativeJavaListTest.java:161

RegexpTest > unsupportedFlagCtor FAILED
    org.junit.ComparisonFailure at RegexpTest.java:71

**Comment by rPraml** [comment]
Fixed the condition in the "filter"method. Now ShellTimerTest fails. Waiting fpr #1672, then I'll do a rebase

**Comment by rbri** [comment]
@rPraml #1672 is merged go ahead :-)

File: rhino/src/main/java/org/mozilla/javascript/Context.java [hunk 1]
```diff
@@ -2568,23 +2569,29 @@ static String getSourcePositionFromStack(int[] linep) {
...
-        }
 
-        return null;
+        return getSourcePositionFromJavaStack(linep);
+    }
+
+    /** Returns the current filename in the java stack. */
+    @SuppressWarnings("AndroidJdkLibsChecker")
+    // Android uses interpreter, so we should not get here.
+    static String getSourcePositionFromJavaStack(int[] linep) {
```
Comment (rPraml) [inline rhino/src/main/java/org/mozilla/javascript/Context.java:2579]: Rebase is done. Maybe someone can confirm, that android always uses interpreter, so we should not get here ;)

Comment (rbri) [reply rhino/src/main/java/org/mozilla/javascript/Context.java:2579]: see @gbrail's comment on #1658

**Comment by rbri** [comment]
@rPraml thanks for that, faster tests are always a good thing

**Pull request merged**